lib/puppet/pops/validation/validator_factory_4_0.rb

```diff
@@ -36,6 +36,8 @@ def severity_producer
     p[Issues::NAME_WITH_HYPHEN]               = :error
     p[Issues::EMPTY_RESOURCE_SPECIALIZATION]  = :ignore
     p[Issues::CLASS_NOT_VIRTUALIZABLE]        = :error
+
+    p[Issues::ILLEGAL_NONLITERAL_PARAMETER_TYPE] = Puppet[:strict] == :off ? :ignore : Puppet[:strict]
```

Comment:
Seems like the least surprising thing would be to let it default to `warning`. In the case there is just a warning would the type just not be enforced? That doesn't seem terribly dangerous?

Reply:
It's a bit subtle. In the case of `Integer[1-3]`, the type is actually valid, because it evaluates to `Integer[-2]`, which in the context of a class parameter `Integer[-2] $i` means it will accept integers >= -2. However, the manifest author intended to specify the range `Integer[1, 3]`. So if you pass a value bigger than 3, then it passes validation (surprise)!